src/index.js

```diff
@@ -949,28 +1064,39 @@ export default function (babel) {
             " with a (key, value) pair.");
           }
 
+          const node = tailPath.node;
+          const n = (...args) => nodeAt(node, ...args);
           const keyExpr = seqExpr.expressions[0];
           const valExpr = seqExpr.expressions[1];
 
-          return t.assignmentExpression(
+          return n("assignmentExpression",
             "=",
-            t.memberExpression(id, keyExpr, true),
+            n("memberExpression", id, keyExpr, true),
             valExpr
           );
         });
 
-        path.replaceWith(wrapComprehensionInIife(id, t.objectExpression([]), path.node.loop));
+        path.replaceWith(
+          wrapComprehensionInIife(
+            id,
+            nodeAt(path.node, "objectExpression", []),
+            path.node.loop
+          )
+        );
       },
 
       TildeCallExpression: {
         // run on exit instead of enter so that SafeMemberExpression
         // can process differently from a wrapping CallExpression
         // eg; `a?.b~c()` -> `a == null ? null : c(a.b)`
         exit(path) {
-          const callExpr = t.callExpression(path.node.right, [
-            path.node.left,
-            ...path.node.arguments,
-          ]);
+          // XXX: this might be breaking source maps because it makes impossible
+          // ordering, right appears after left in original source but before
```

Comment:
yeah, both third party tools choking and developers getting confused I think are the potential issues. 

But frankly I'm happy to ship something that does a halfway decent job for now and try to get more accurate later, all up to you on that front I think. 

Biggest use-case in my mind would be Flow, where we're going to give flow a transformed AST, and it's going to see a type error, and shoot back the error with its loc to the editor, which will display a squiggly red line at that loc... 

It's kind of hard to audit / think through this in github code review without experiencing it myself